[JIT] Enable saving modules with hooks in FBCODE

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51241 [JIT] Enable saving modules with hooks in FBCODE**

Differential Revision: [D26111488](https://our.internmc.facebook.com/intern/diff/D26111488)

Remove guard against saving modules with hooks in FBCODE. Guards originally added for forward compatibility issue where older PyTorch deserialization binaries couldn't handle serialized modules with hooks attached. Should land this guard removal once the older binaries are no longer in FB prod.

Commit:
[JIT] Enable saving modules with hooks in FBCODE

[ghstack-poisoned]

SplitInfinity reviewed (approved):
🚀 

Should we enable all the unit tests in fbcode too? As I recall we disabled some of them.

Commit:
Update on "[JIT] Enable saving modules with hooks in FBCODE"


Differential Revision: [D26111488](https://our.internmc.facebook.com/intern/diff/D26111488)

Remove guard against saving modules with hooks in FBCODE. Guards originally added for forward compatibility issue where older PyTorch deserialization binaries couldn't handle serialized modules with hooks attached. Should land this guard removal once the older binaries are no longer in FB prod.  

[ghstack-poisoned]
